core/frontend/src/libs/MAVLink2Rest/index.ts

```diff
@@ -104,6 +104,11 @@ class Mavlink2RestManager {
    *
    */
   sendMessageViaWebsocket(message: any): void {
+    if (!this.socket) {
+      this.socket = this.createSocket(this.baseUrl)
+      this.socket.onopen = () => this.sendMessageViaWebsocket(message)
+      return
```

Comment:
That's a weird case, for this to be the case, the constructor would not be called at all, because that is how `createSocket` happens.

How `socket` gets undefined since the usage of the class itself makes it exist ?
And if it's closed, https://github.com/bluerobotics/BlueOS-docker/pull/1437/files#diff-df3bf14ffcc5f9af5bdd02a4c9f2ad4d7f85532d34d2ce3b095e8d0bafa7892aR82 should take effect.

Reply:
`probeBaseUrlCandidates` is failiting on my end in the `fetch(asHtto)` step, which means `setBaseUrl` is never called.

Reply:
Doesn't it make more sense to fix it under `probeBaseUrlCandidates` ? Where if the probe is failing it reeschedule to run again ? Or add a fallback with `this.createSocket(this.baseUrl)` ? That's the same thing that you are doing.

Reply:
this already cost me more time than I wish it did, I'll come back to it later.